Sources/Workspace/InitPackage.swift

```diff
@@ -350,8 +378,58 @@ public final class InitPackage {
         }
     }
 
+    private func writePlugins() throws {
+        switch packageType {
+        case .buildToolPlugin, .commandPlugin:
+            let plugins = destinationPath.appending(component: "Plugins")
+            guard self.fileSystem.exists(plugins) == false else {
+                return
+            }
+            progressReporter?("Creating \(plugins.relative(to: destinationPath))/")
+            try makeDirectories(plugins)
+
+            let moduleDir = plugins.appending(component: "\(pkgname)")
+            try makeDirectories(moduleDir)
```

Comment:
do we want to create a flat structure here like we have done elsewhere?

cc @bitjammer @TimTr

Reply:
Seemed like we only did that for executables and libraries, I think... which is why I followed suit here.  But happy to change it.

Reply:
I was wrong.  Here's the current code:
```
        switch packageType {
        case .executable, .tool:
            moduleDir = sources
        default:
            moduleDir = sources.appending("\(pkgname)")
        }
```
So we only do the flat structure for the two forms of executable today, not for library or compiler macro.  Should we do it for all of them?

Reply:
@abertelrud Correct, just for the executable so far. However, with the changes in https://github.com/apple/swift-package-manager/pull/6294 it is now default behavior for SwiftPM to work properly with just `/Sources/` if there is only one target, or also search through specific target folders with proper naming when there is more than one. No need to put the `path:` command in either case. So it is now a bit easier to start simple and add complexity as needed.

So for this case, the primary question is if we think multi-target is very likely, very early for these templates. Or if many folks will happily have a single build product, then nice to default the template to the simple, flat layout. It is now pretty easy to add the grouping later, without any special `Package.swift` code.

Reply:
Great, didn't know about that PR.  Sure, I'll make them flat.  Not having to have extra stuff in the manifest was one of the arguments against.

Reply:
Updated with flat structure.